MapboxDirections/MBLane.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+
+@objc(MBLaneIndication)
+public enum LaneIndication: Int, CustomStringConvertible {
+    
+    /**
+     An indication indicating a turn to the left.
+     */
+    case Left
+    
+    /**
+     An indication indicating a turn to the right.
+     */
+    case Right
+    
+    /**
+     An indication indicating a sharp turn to the left.
+     */
+    case SharpLeft
+    
+    /**
+     An indication indicating a sharp turn to the right.
+     */
+    case SharpRight
+    
+    /**
+     An indication indicating a slight turn to the left.
+     */
+    case SlightLeft
+    
+    /**
+     An indication indicating a slight turn to the right.
+     */
+    case SlightRight
+    
+    /**
+     No dedicated indication is shown.
+     */
+    case StraightAhead
+    
+    /**
+     An indication signaling the possibility to reverse
+    */
+    case Uturn
+    
+    /**
+     No dedicated indication is shown.
+    */
+    case None
+    
+    public init?(description: String) {
+        let type: LaneIndication
+        switch description {
+        case "left":
+            type = .Left
+        case "right":
+            type = .Right
+        case "sharp left":
+            type = .SharpLeft
+        case "sharp right":
+            type = .SharpRight
+        case "slight left":
+            type = .SlightLeft
+        case "slight right":
+            type = .SlightRight
+        case "straight":
+            type = .StraightAhead
+        case "uturn":
+            type = .Uturn
+        case "none":
+            type = .None
+        default:
+            return nil
+        }
+        self.init(rawValue: type.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .Left:
+            return "left"
+        case .Right:
+            return "right"
+        case .SharpLeft:
+            return "sharp left"
+        case .SharpRight:
+            return "sharp right"
+        case .SlightLeft:
+            return "slight left"
+        case .SlightRight:
+            return "slight right"
+        case .StraightAhead:
+            return "straight"
+        case .Uturn:
+            return "uturn"
+        case .None:
+            return "None"
+        }
+    }
+}
+
+@objc(MBLane)
+public class Lane: NSObject {
+    
+    public let indications: [LaneIndication]
```

Comment:
It occurs to me that we need to treat this as an option set rather than an array of enum values. After all, a lane can’t be `[.Uturn, .Uturn]`. Option sets are really nice in Objective-C and really nice in Swift, but unfortunately Swift option sets don’t bridge to Objective-C. So you’ll have to implement LaneIndication in Objective-C. Use MapboxGeocoder.swift’s [`MBPlacemarkScope`](https://github.com/mapbox/MapboxGeocoder.swift/blob/master/MapboxGeocoder/MBPlacemarkScope.h) option set and [`PlacemarkScope`](https://github.com/mapbox/MapboxGeocoder.swift/blob/master/MapboxGeocoder/MBPlacemarkScope.swift) extension as a model.